ibtk/include/ibtk/HierarchyTimeInterpolator.h

```diff
@@ -0,0 +1,226 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2022 - 2022 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBTK_HierarchyInterpolator
+#define included_IBTK_HierarchyInterpolator
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibtk/SnapshotCache.h>
+#include <ibtk/ibtk_utilities.h>
+
+#include "CellVariable.h"
+#include "HierarchySideDataOpsReal.h"
+#include "IntVector.h"
+#include "SideVariable.h"
+#include "tbox/Database.h"
+#include "tbox/Pointer.h"
+
+#include <string>
+#include <vector>
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBTK
+{
+/*!
+ * \brief Class HierarchyTimeInterpolator provides a method of determining and storing average fields over periodic
+ * intervals. This class can also be used to determine intermediate velocity fields when at a periodic steady state.
```

Comment:
Since this is doing more than just interpolating data in time, I'd suggest considering a more specific class name. I am not crazy about these suggestion, but something like `HierarchyAveragedDataManager` or `AveragedSnapshotDataManager`?

This might be too complicated, but you could also make a little class hierarchy --- `SnapshotTimeInterpolator` or `SnapshotDataManager` for interpolating snapshots in time, and a subclass like `SnapshotMeanTimeInterpolator` or `SnapshotMeanDataManager` for tracking mean values.

Reply:
We actually don't need a class to interpolate between snapshots. It would probably be cleaner to make those stand alone functions.